patches/tModLoader/Terraria/ModLoader/UI/ModBrowser/ModDownloadNotification.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria.GameContent;
+using Terraria.GameInput;
+using Terraria.ModLoader.UI.DownloadManager;
+using Terraria.UI;
+
+namespace Terraria.ModLoader.UI.ModBrowser;
+
+public class ModDownloadNotification : IInGameNotification, IDownloadProgress
+{
+	// Remove this notification once the download completes and then the 3-second timer is up.
+	public bool ShouldBeRemoved => timeLeft <= 0;
+
+	private int timeLeft = 3 * 60;
+
+	private int animationTimer = 0;
+
+	private string message;
+	private ModDownloadItem downloadItem;
+
+	public ModDownloadNotification(ModDownloadItem downloadItem)
+	{
+		this.downloadItem = downloadItem;
+		message = $"Download pending {downloadItem.DisplayName}";
+	}
+
+	private float Scale {
+		get {
+			if (timeLeft < 30) {
+				return MathHelper.Lerp(0f, 1f, timeLeft / 30f);
+			}
+
+			if (timeLeft > 165) {
+				return MathHelper.Lerp(1f, 0f, (timeLeft - 165) / 15f);
+			}
+
+			return 1f;
+		}
+	}
+
+	private float Opacity {
+		get {
+			if (Scale <= 0.5f) {
+				return 0f;
+			}
+
+			return (Scale - 0.5f) / 0.5f;
+		}
+	}
+
+	public void Update()
+	{
+		animationTimer++;
+
+		if (!downloadItem.IsInstalled) {
+			if (timeLeft > 120)
+				timeLeft--;
+			return;
+		}
+
+		timeLeft--;
+
+		if (timeLeft < 0) {
+			timeLeft = 0;
+		}
+	}
```

Comment:
some of the UI/counter behaviour could/should definitely be generified. The `ModDownloadNotification` subclass should only have to provide information on when to transition between UI states (eg 'still downloading so must persist', 'finished downloading so can last for 3s/be cleared') and drawing